torch/_dynamo/variables/builder.py

```diff
@@ -768,6 +769,35 @@ def build_key_value(i, k, v):
             var = TorchFunctionModeVariable(value, source=self.source)
             self.tx.output.side_effects.track_object_existing(value, var)
             return var
+        elif istype(value, set):
+            if any(isinstance(x, torch.Tensor) for x in value):
+                unimplemented_v2(
+                    gb_type="Attempted to wrap a set with tensors",
+                    context="Python set containing torch.Tensor elements",
+                    explanation=(
+                        "Dynamo cannot trace sets of tensors. To get a stable ordering, "
+                        "Dynamo needs to convert the set into a list and the order might not be "
+                        "stable if the set contains tensors."
+                    ),
+                    hints=["Use a dictionary instead", *graph_break_hints.SUPPORTABLE],
+                )
```

Comment:
@zou3519, I've updated the code to graph break if the set contains a torch tensor.

Reply:
nit: use a dictionary where the keys are tensors.